### PR TITLE
i2c: impl. generics

### PR DESCRIPTION
**Pull request by xoviat**

### CODE REVIEW

File: src/i2c.rs [hunk 1]
```diff
@@ -556,6 +556,54 @@ pub enum Error {
     ARBITRATION,
 }
 
+mod private {
+    pub trait Sealed {}
+}
+
+// Implemented by all USART instances
```
Comment (xoviat) [inline src/i2c.rs:563]:
```suggestion
// Implemented by all I2C instances
```

**Comment by xoviat** [comment]
@thalesfragoso this isn't too urgent, but if that's the only problem with this, I'll squash the commits, and you can merge it.

**Comment by therealprof** [comment]
Looks like a good improvement, although I'm not too excited about the use of macros here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Change `WriteBuffer + 'static` to `StaticWriteBuffer`in the DMA module.
 - Fixed a race condition where SPI writes could get stuck in an error state forever (PR #269).
 - Implement generics on the serial module.
+- Implement genercis on the i2c module, not including fast i2c.
```
Comment (xoviat) [inline CHANGELOG.md:32]:
```suggestion
- Implement generics on the i2c module, not including fast i2c.
```

**Comment by thalesfragoso** [comment]
I think the macros are okay in this case, they aren't implementing much functionality anyways, just enabling clocks and providing a backwards compatible function.

**Comment by therealprof** [comment]
@thalesfragoso I'm generally no fan of macros, especially if they don't do much. 😅🤷🏻‍♂️

**Review by thalesfragoso — DISMISSED** [review]

**Review by therealprof — APPROVED** [review]
LGTM, thanks.

bors r+

**Comment by xoviat** [comment]
cc @therealprof

**Review by therealprof — APPROVED** [review]
bors r+

**Pull request merged**